### DISCUSSION TITLE
Can Open-OCPP stack be used on ESP32 controller

### DISCUSSION THREAD

**Discussion by Bhuvi-Naveen** [c-jimenez/open-ocpp#127] [General]
Hi,

Thanks to confirm, if we could use open-ocpp stack on ESP32.

What will be the main challenges?

Regards,

**Comment by c-jimenez** [comment]
Hi,

I don't think that its not possible...but it will be challenging :)

The 2 main dependencies which may be hard to use on ESP32 are :
- SQLite3
- OpenSSL

If you can have an embedded filesystem, you'll be able to compile sqlite3 against it and make the persistency work "as-is". If you don't have any filesystem, you'll have to write a new Database class which emulates it.

For OpenSSL it will be much difficult. The easiest part is to compile the libwebsocket library against MBedTLS instead OpenSSL.
The most challenging part is to port all the classes from the [tools/x509](https://github.com/c-jimenez/open-ocpp/tree/develop/src/tools/x509) directory from OpenSSL to MBedTLS.

If you achieve this port, I'll be pleased to review it since it may benefit to a lot of people working with microcontrollers.

**Comment by Bhuvi-Naveen** [comment]
Hi,

Thank you c-jimenez for responding back!

I will keep posted.

regards,

**Comment by SavaAlexandru** [comment]
Hello. Thanks for the last reply. I have a question:
Did anything change for integrating with a MCU ?
